keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -0,0 +1,1009 @@
+
+#### Properties
+
+Name/value pairs to represent properties of the clusters. It could be a
+collection of ClusterProperty resources, but could also be info based on
+other implementations.
+
+#### Conditions
+
+Record cluster’s healthiness condition and easy to extend, conforming to
+metav1.Condition format.
+
+Predefined condition types:
+- Healthy conditions indicate the cluster is in a good state. (state:
+  True/False/Unknown). Healthiness can have different meanings in
+  different scenarios. We will have multiple condition types to define
+  healthiness:
+  - ControlPlaneHealthy is to define whether the controlplane of the
+    cluster is in the healthy state.
+    * apiserver/healthz
+    * controller-manager/healthz
+    * scheduler/healthz
+    * etcd/healthz
+  - <<[UNRESOLVED]>> AllNodesHealthy is to define if the nodes in the cluster are in a
+    healthy state. If one node is not healthy, the status of NodeHealthy
+    is set to false with the message indicating details. (todo
+    tolerance, it should be configurable)
+  It would be useful to collect the healthiness of other subsystems in
+  the cluster, e.g. network, dns, storage, or ingress. However, it is
+  not easy to collect that information in a common way with different
+  implementations of network or storage providers. We decide not to
+  include other subsystems healthiness conditions in the initial phase.
+- Joined: indicate the cluster is under management by the cluster manager.
+  The status of the cluster SHOULD be updated by the cluster manager under
+  this condition.
+- Schedulable: define if the cluster is eligible to be scheduled the workload. By
+  default, if at lease one node in the cluster is schedulable, the cluster will be set
+  as schedulable. In the future, we would consider a spec field to define the criteria
+  of schedulable status.
+
+## API Example
+
+```yaml
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ClusterInventory
+metadata:
+ name: generated-cluster-name
+ labels:
+   x-k8s.io/cluster-manager: some-cluster-manager
+spec:
+  displayName: some-cluster
+  clusterManager:
+    name: some-cluster-manager
+status:
+ version:
+   kubernetes: 1.28.0
+ properties:
+   - name: clusterset.k8s.io
+     value: some-clusterset
+   - name: location
+     value: apac
+ conditions:
+   - type: ControlPlaneHealthy
+     status: True
+     lastTransitionTime: "2023-05-08T07:56:55Z"
+     message: ""
+   - type: AllNodesHealthy
+     status: True
+     lastTransitionTime: "2023-05-08T07:57:55Z"
+     message: ""
+   - type: Joined
+     status: True
+     lastTransitionTime: "2023-05-08T07:58:55Z"
+     message: ""
+   - type: Schedulable
+     status: True
+     lastTransitionTime: "2023-05-08T07:58:55Z"
+     message: ""
+```
+
+### Scalability implication
+
+The API should provide summarized metadata of the cluster and relatively "static" cluster status.
+Dynamic data, e.g. cluster resource usage, should not be included in this API give it will bring
```

Comment:
Should the bullet up above in the Design section where there is a user question:

> Does the cluster have sufficient resources?

Be adjusted to specifically imply that its Boolean or something less variable then the percentage or available value

Reply:
The wording of the scalability section is updated.